adafruit_midi/midi_message.py

```diff
@@ -285,6 +285,20 @@ def from_bytes(cls, msg_bytes):
         representation of the MIDI message."""
         return cls()
 
+    def __str__(self):
+        """Print an instance"""
+        cls = self.__class__
+        if slots := getattr(cls, "_message_slots", None):
```

Comment:
yes, walrus operator works at least from 7.3 forward.

no, using `self.__dict__` didn't seem workable (nor `dir`). For instance, because `Note.channel` is a property backed by `Note._channel`, using `self.__dict__.items()` would make the repr look like:
```
NoteOn(note=3, velocity=127, _channel=None)
```